.github/workflows/publish-pypi.yaml

```diff
@@ -0,0 +1,96 @@
+name: Publish to PyPI
+
+on:
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - '**'
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Upload release to PyPI
+    needs:
+    - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/suspect
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
```

Comment:
Does this need an added secret to make this work? I haven't done much with GHA.

Reply:
This is the first time for me as well on setting up the PyPI publishing from GHA.

From reading the link below, it uses Open ID Connect which we need to set up in PyPI's trusted publishing settings page.

https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi#adding-the-identity-provider-to-pypi